### PR TITLE
Dash opioid epidemic

### PR DESCRIPTION
**Pull request by YunkeXiao**
Issue for app: #29

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version): https://dash-playground.plotly.host/dash-opioid-epidemic/
- Current gallery app URL: https://dash-gallery.plotly.host/dash-opioid-epidemic/

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [x] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [x] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

**Comment by shammamah** [comment]
@YunkeXiao Please fill out the first line of the PR comment where it says "Issue for app".

**Comment by shammamah** [comment]
Please also fix the mobile layout!

**Comment by OwenMatsuda** [comment]
For better code readability:
You repeat a lot of code with things like figure["layout"] or fig["data"], so you could make an additional variable to avoid this. Other than that, looks good!

**Comment by shammamah** [comment]
@YunkeXiao Please center the header text for the mobile layout. Other than that, it looks good to me :)

**Comment by shammamah** [comment]
@YunkeXiao Could you please change the size of the "Deaths are classified..." text in the mobile layout to be a bit smaller? Other than that, 💃

**Comment by ronniebugia** [comment]
image

**Comment by shammamah** [comment]
Something weird is happening with the graph title in the mobile view:
Screen Shot 2019-07-16 at 12 16 30 PM

**Comment by YunkeXiao** [comment]
> Something weird is happening with the graph title in the mobile view:
> Screen Shot 2019-07-16 at 12 16 30 PM
How did you get that? Don't have anything wrong on my end if I reload.

**Comment by shammamah** [comment]
@YunkeXiao Getting an application error on Playground -- please fix and then I can review!

File: apps/dash-opioid-epidemic/app.py [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+        )
+
+        fig_layout = fig["layout"]
+        fig_data = fig["data"]
+
+        fig_data[0]["text"] = deaths_or_rate_by_fips.values.tolist()
+        fig_data[0]["marker"]["color"] = "#2cfec1"
+        fig_data[0]["marker"]["opacity"] = 1
+        fig_data[0]["marker"]["line"]["width"] = 0
+        # TODO: Why doesn't the text show up over the bars?
```
Comment (ronniebugia) [inline apps/dash-opioid-epidemic/app.py]: Are you going to address this TODO ?

Comment (ronniebugia) [reply apps/dash-opioid-epidemic/app.py]: If not you can just remove this line

**Pull request merged**